src/MainWindow.cpp

```diff
@@ -2540,3 +2542,49 @@ sqlb::ObjectIdentifier MainWindow::currentlyBrowsedTableName() const
                                   ui->comboBrowseTable->currentData(Qt::EditRole).toString());  // Use the edit role here to make sure we actually get the
                                                                                                 // table name without the schema bit in front of it.
 }
+
+void MainWindow::on_actionHideColumns_triggered()
+{
+     int myCol = -1;
+     if(ui->dataTable->selectedCols().size() == 0)
+     {
+          myCol = ui->actionBrowseTableEditDisplayFormat->property("clicked_column").toInt();
+     }
+
+     foreach(int col, ui->dataTable->selectedCols())
+     {
+          ui->dataTable->hideColumn(col);
+     }
+     if(myCol != -1)
+          ui->dataTable->hideColumn(myCol);
+
+     // check to see if all the columns are hidden
+
+     bool allHidden = true;
+     for(int col = 1; col < ui->dataTable->model()->columnCount(); col++)
+     {
+          if(!ui->dataTable->isColumnHidden(col))
+          {
+                allHidden = false;
+                break;
+          }
+     }
+
+     if(allHidden  && ui->dataTable->model()->columnCount() > 1)
+     {
+          ui->dataTable->showColumn(1);
+          ui->dataTable->resizeColumnToContents(1);
+     }
```

Comment:
Can you add an entry to ```struct BrowseDataTableSettings``` in ```MainWindow.h``` for saving the hidden flag of each column? It would more or less need to be implemented and used like the ```columnWidths``` member.

This would make the hidden columns stay hidden when changing to another table and coming back to the original table later. It would also save the hidden columns in the project files.

If you're unsure about this, that's no problem either. We can just merge the commit then and I'll add this later :smile:

Reply:
If you could add it in later, that would be great!

I don't think I can really dig into the full column-width persistence in the near future.

Just FYI about that usage... I know that when you say `showColumn()` in Qt 4, `QTableView` doesn't respect previous column sizes for all but the most recently hidden column.  Any after the first column that have been hidden, get a width of zero.  It took me a while to figure that one out... this post on stackoverflow that helped point out the issue for me. https://stackoverflow.com/a/7595202/999943

Reply:
Cool, thanks for the information 👍 That would have taken me some time as well :wink: